### PR TITLE
Start fixing errors when enabling 'strictNullChecks' in office-ui-fabric-react

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

I'm working on enabling 'strictNullChecks' in office-ui-fabric-react. There are currently over 800 errors when enabling this flag, so I'll be fixing this in stages. There are many legitimate bugs exposed by this change, but not all of them are safe to fix, so most fixes in this branch as simply typecasts. The goal is to enable the compiler option in order to prevent new issues from entering the code base. 

For those not familiar with the ! appearing at the end of variables, it's the [non-null assertion operator](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export interface IComboBoxState {
   isOpen?: boolean;
 
   // The currently selected index (-1 if no index is selected)
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: @jspurlin, there appears to be some legitimate bugs in ComboBox exposed by enabling strictNullChecks. There appears to be some situations where we aren't handling null and undefined values correctly. These changes should be functionally equivalent to the previous combo box for the most part, but we should go back in here and address these issues by either being more strict in our types (not allowing null/undefined for some types, removing some optionals, or properly handling null/undefined)

Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: @christiango Sounds good, feel free to open a bug and assign it to me

File: packages/office-ui-fabric-react/src/components/Persona/Persona.tsx [hunk 2]
```diff
@@ -98,12 +97,13 @@ export class Persona extends BaseComponent<IPersonaProps, IPersonaState> {
       onRenderOptionalText,
       imageShouldStartVisible
     } = this.props;
+    let size = this.props.size as PersonaSize;
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.tsx]: Should variable creation be using static typing  (e.g. ':' type assignment) as opposed to 'as' casting?

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx [hunk 3]
```diff
@@ -302,7 +302,7 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
     const element: HTMLInputElement = event.target as HTMLInputElement;
     const value: string = element.value;
     if (this.state.value) {
-      const newValue = this._onValidate(value);
+      const newValue = this._onValidate && this._onValidate(value);
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: Should this be this._onValidate!(value) ?

File: packages/office-ui-fabric-react/src/components/Persona/Persona.tsx (hunk 2 above)
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Persona/Persona.tsx]: The type is inferred here. We typically don't specify the type when the compiler is able to figure it out (usually the case when doing an assignment with an initial value). Doing size: PersonaSize would still need a typecast anyway to get around strictNullChecks.

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx (hunk 3 above)
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx]: Yes it should. Nice catch.

**Pull request merged**